listretriever.go

```diff
@@ -81,7 +81,13 @@ func (gh gitHubListRetriever) GetLatestReleaseTag() (string, error) {
 func (gh gitHubListRetriever) GetList(release string) (io.Reader, error) {
 	var url = fmt.Sprintf(publicSuffixURL, release)
 
-	var res, err = gh.client.Get(url)
+	// Just in case a nil client was passed, use the default http client.
+	client := http.DefaultClient
```

Comment:
This might be needed in the GetLatestReleaseTag()  function as well.

Reply:
Ignore me your passing it in UpdateWithListRetriever

Reply:
I am confused why do you need this for GetList() and not for GetLatestReleaseTag()

Reply:
Good catch! I just updated it. Also added a method to retrieve the client.

Reply:
Just a safeguard!